### PR TITLE
Delete filtered run results

### PR DESCRIPTION
**Pull request by csordasmarton**

### CODE REVIEW

File: tests/functional/report_viewer_api/test_remove_run_results.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
```
Comment (gyorb) [inline tests/functional/report_viewer_api/test_remove_run_results.py]: Please add Module docstring and `__future__` imports.

File: libcodechecker/libclient/thrift_helper.py [hunk 2]
```diff
@@ -70,6 +70,10 @@ def getRunResultTypes(self, runId, reportFilters):
     def removeRunResults(self, run_ids):
```
Comment (gyorb) [inline libcodechecker/libclient/thrift_helper.py]: Shouldn't we extend the helper with the new function arguments?
```def removeRunResults(self, run_ids, report_filter, cmp_data):```

File: api/v6/report_server.thrift [hunk 3]
```diff
@@ -383,9 +383,16 @@ service codeCheckerDBAccess {
 
   // remove bug results from the database
   // PERMISSION: PRODUCT_STORE
```
Comment (gyorb) [inline api/v6/report_server.thrift]: Please extend the API description with a Warning about the behavior change in the API call. (Will not remove the run only results will be deleted and use the new api call removeRun to actually delete a run with the results together.

File: api/v6/report_server.thrift [hunk 4]
```diff
@@ -382,10 +382,24 @@ service codeCheckerDBAccess {
...
+  // remove bug results from the database
+  // PERMISSION: PRODUCT_STORE
+  bool removeRunReports(1: list<i64>    runIds,
+                        2: ReportFilter reportFilter,
+                        3: CompareData  cmpData)
                         throws (1: shared.RequestFailed requestError),
 
+  // Remove run from the database.
+  // PERMISSION: PRODUCT_STORE
+  bool removeRun(1: i64 runId)
```
Comment (gyorb) [inline api/v6/report_server.thrift]: Do we want to support of the removal of multiple runs through this API?

File: libcodechecker/server/api/report_server.py [hunk 5]
```diff
@@ -1758,51 +1758,108 @@ def removeRunResults(self, run_ids):
...
+                return False
 
-        return not failed
+    @exc_to_thrift_reqfail
+    @timeit
+    def removeRun(self, run_id):
+        self.__require_store()
+
+        # Remove all reports from the run.
+        ret = self.removeRunReports([run_id], None, None)
```
Comment (gyorb) [inline libcodechecker/server/api/report_server.py]: Do we need to remove the run reports separately before removing the run?
The cascade delete should remove the reports for a run.

File: api/v6/report_server.thrift (hunk 4 above)
Comment (csordasmarton) [reply api/v6/report_server.thrift]: I don't think so. This way we can call this function asynchronously and we can handle warnings much better for example if a run is locked.

**Pull request merged**